bundle-workflow/src/test_workflow/dependency_installer.py

```diff
@@ -0,0 +1,65 @@
+import os
+import shutil
+
+from aws.s3_bucket import S3Bucket
+
+
+class DependencyInstaller:
+    """
+    Provides a dependency installer for the test suites.
+    """
+
+    def __init__(self, build):
+        self.build_id = build.id
+        self.version = build.version
+        self.arch = build.architecture
+        self.s3_maven_location = (
+            f"/builds/{self.version}/{self.build_id}/{self.arch}/maven/org/opensearch"
+        )
+        self.s3_build_location = (
+            f"/builds/{self.version}/{self.build_id}/{self.arch}/plugins"
+        )
+
+    def install_maven_dependencies(self, dependency_dict):
+        """
+        Downloads the maven dependencies from S3 and puts them on the maven local path
+        for each dependency in the dependency_list.
+
+        :param dependency_dict: list of dependency names with version for which the maven artifacts need to be downloaded.
+        Example: {'opensearch-job-scheduler':'1.1.0.0', 'opensearch-core':'1.1.0'}
+        """
+        s3_bucket = S3Bucket(self.s3_maven_location)
+        for dependency, version in dependency_dict.items():
+            s3_path = f"{dependency}/{version}"
+            maven_local_path = os.path.join(
+                os.path.expanduser("~"),
+                f".m2/repository/org/opensearch/{dependency}/{version}/",
+            )
+            s3_bucket.download_folder(s3_path, maven_local_path)
```

Comment:
nit: this is s3_relative_path

Reply:
It should be s3_maven_location for the dependency  which is a relative path inside the bucket. An example -  `builds/1.1.0/24/x64/maven/org/opensearch/`